### PR TITLE
fix version line parser

### PR DESCRIPTION
**Pull request by AccraZed**
## Summary

This addresses the issue outlined [here](https://github.com/paketo-buildpacks/dotnet-core/issues/440), where it seems like the version line parser was allowing certain lines that shouldn't have been passing.
## Use Cases

This change should allow users to get more direct error messages when building their apps with custom framework versions
## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Review by fg-j — CHANGES_REQUESTED** [review]
Can you please add a unit test that fails without this fix and then passes once this fix is added? This will a) help "document" the purpose of the regex and b) ensure we have good unit test coverage here.

**Comment by AccraZed** [comment]
The test case has one concern regarding the specificity of the error - currently, the [version parse function](https://github.com/paketo-buildpacks/dotnet-execute/blob/main/project_file_parser.go#L79) returns an error detailing that the TargetFramework property is missing, when there can be cases of malformed version lines in otherwise valid TargetFramework properties that gets errored as missing, instead of something more specific. 

Maybe the error message could be generalized to reference both kinds of errors, or another check can be used to differentiate missing from malformed version lines?

File: project_file_parser.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (p ProjectFileParser) ParseVersion(path string) (string, error) {
 
 	// This regular expression matches on 'net<x>.<y>',
 	// 'net<x>.<y>-<platform>' & 'netcoreapp<x>.<y>'
-	targetFrameworkRe := regexp.MustCompile(`net(?:coreapp)?(\d\.\d)(?:\w+)?`)
+	targetFrameworkRe := regexp.MustCompile(`net(?:coreapp)?(?:(\d\.\d)(?:\-?\w+)?)$`)
 	for _, group := range project.PropertyGroups {
 		matches := targetFrameworkRe.FindStringSubmatch(group.TargetFramework)
 		if len(matches) == 2 {
```
Comment (fg-j) [inline project_file_parser.go:74]: Taking a look at this loop that iterates over the property groups in the XML file, it seems like the error we currently return is inaccurate/misleading. We don't directly check for the presence of the `<TargetFramework>` tag. Maybe rewording the error to say something like "missing _or invalid_ TargetFramework property" would help clarify things.

**Review by fg-j — APPROVED** [review]
Confirmed that the buildpack now properly fails detection and returns an informative error when the target framework version is of the (invalid) form `netcoreappX.Y.Z`
